### PR TITLE
pacman: fix changed status when ignorepkg has been defined

### LINKED ISSUE #1758: pacman does not honor the "ignore" settings in package diff
**Summary**

I have an ignore setting on one of my systems, telling it to ignore any php upgrade. When I then upgrade said system the diff shows that it upgraded php, but it doesn't actually upgrades php. Using the playbook I check which services need to be restarted to use new versions and it always restarts `php-fpm` in this case.

**Issue Type**

Bug Report

**Component Name**

pacman

**Ansible Version**

```
ansible 2.9.17
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/mschlapa/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.8/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.8.7 (default, Jan 20 2021, 00:00:00) [GCC 10.2.1 20201125 (Red Hat 10.2.1-9)]
```

**Configuration**

```paste below

```

**OS / Environment**

ArchLinux

**Steps To Reproduce**

`/etc/pacman.conf`
```
# Pacman won't upgrade packages listed in IgnorePkg and members of IgnoreGroup
IgnorePkg   = php php-apcu php-fpm php-gd php-imagick php-imap php-intl imagemagick
#IgnoreGroup =
```

**Expected Results**

No diff.

**Actual Results**

```diff
--- before
+++ after
@@ -1,11 +1,11 @@
-imagemagick-7.0.10.60-1
-php-7.4.14-1
-php-apcu-5.1.19-1
-php-fpm-7.4.14-1
-php-gd-7.4.14-1
-php-imagick-3.4.4-8
-php-imap-7.4.14-1
-php-intl-7.4.14-1
+imagemagick-7.0.10.62-1
+php-8.0.2-1
+php-apcu-5.1.19-2
+php-fpm-8.0.2-1
+php-gd-8.0.2-1
+php-imagick-3.4.4.r66.g448c1cd-2
+php-imap-8.0.2-1
+php-intl-8.0.2-1
```

### PR DESCRIPTION
**Pull request by aminvakil**
##### SUMMARY
Fixes #1758 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
pacman

### CODE REVIEW

**Comment by aminvakil** [comment]
@Schrottfresse Since this module does not have any unit/integration tests, could you please check if this PR solves the issue?
I've tested this locally and it's working fine as intended.
You can also use document on [testing on PR locally](https://github.com/ansible/community-docs/blob/main/test_pr_locally_guide.rst).

**Comment by Schrottfresse** [comment]
Yep, PR works for me!

File: plugins/modules/packaging/os/pacman.py [hunk 1]
```diff
@@ -254,16 +254,23 @@ def upgrade(module, pacman_path):
         # e.g., "ansible 2.7.1-1 -> 2.7.2-1"
         regex = re.compile(r'([\w+\-.@]+) (\S+-\S+) -> (\S+-\S+)')
         for p in data:
-            m = regex.search(p)
-            packages.append(m.group(1))
-            if module._diff:
-                diff['before'] += "%s-%s\n" % (m.group(1), m.group(2))
-                diff['after'] += "%s-%s\n" % (m.group(1), m.group(3))
+            if 'ignored' not in p:
```
Comment (felixfontein) [inline plugins/modules/packaging/os/pacman.py:257]: What happens if `ignored` is part of a package name?

Comment (aminvakil) [reply plugins/modules/packaging/os/pacman.py]: Nice catch! Fixed in https://github.com/ansible-collections/community.general/pull/2936/commits/d55b0c1336b6be5bfcea7feada5b60f51b9f95d8.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@aminvakil thanks for fixing this!
@Schrottfresse thanks for testing!

**Pull request merged**